daemon/mgr/network.go

```diff
@@ -149,26 +150,86 @@ func (nm *NetworkManager) List(ctx context.Context, labels map[string]string) ([
 	return net, nil
 }
 
-// Get returns the information of network that specified name/id.
-func (nm *NetworkManager) Get(ctx context.Context, name string) (*types.Network, error) {
+// GetNetworkByName returns the information of network that specified name.
+func (nm *NetworkManager) GetNetworkByName(name string) (*types.Network, error) {
 	n, err := nm.controller.NetworkByName(name)
-	if err != nil {
-		if err == libnetwork.ErrNoSuchNetwork(name) {
-			return nil, errors.Wrap(errtypes.ErrNotfound, err.Error())
-		}
-		return nil, err
-	}
-
-	if n != nil {
+	if err == nil {
 		return &types.Network{
-			Name:    name,
+			Name:    n.Name(),
 			ID:      n.ID(),
 			Type:    n.Type(),
 			Network: n,
 		}, nil
 	}
+	return nil, err
+}
 
-	return nil, nil
+// GetNetworksByID returns a list of networks that specified id.
+func (nm *NetworkManager) GetNetworksByID(id string) []*types.Network {
+	var matchedNetworks []*types.Network
+
+	walker := func(nw libnetwork.Network) bool {
+		if strings.HasPrefix(nw.ID(), id) {
```

Comment:
I don't know why it appears the same prefix id of networks? Id is not uniquely identifies?

Reply:
@rudyfly 
Thanks a lot for your reviews.
`id`  represents prefix of network ID, or full network ID, and `id` is renamed `partialID`.
Sorry for making you confused.